core/logic.py

```diff
@@ -0,0 +1,150 @@
+from django.utils import timezone
+
+from .api import Api
+from .models import Course
+from .models import User
+from .models import UserCourse
+
+CORE_ERROR_LOG_PATH = "./.yvideo-py-core-error.log"
+
+
+def log_error(error_message, error_origin, error_info={}, exception=None):
+    """
+    Writes an error to the specified error log path. Includes datetime error is reported and
+    which function generated the error. If a python exception is provided, this is also logged.
+    """
+
+    error_time = timezone.now()
+    with open(CORE_ERROR_LOG_PATH, "a") as error_log:
+        error_log.write(
+            f"{error_message}\nfrom {error_origin}\nTime: {error_time}\nError information: {error_info}\n"
+            + ""
+            if exception is None
+            else f"Exception: {exception}\n\n"
+        )
+
+
+def check_for_user_in_db(byu_id):
+    try:
+        user = User.objects.get(byu_id=byu_id)
+        return user
+    except Exception:
+        return None
+
+
+def check_or_create_course(course):
+    # check if a course exists, if it doesn't, create it
+    try:
+        course_obj = Course.objects.filter(
+            dept=course["teaching_area"],
+            catalog_number=course["catalog_number"] + course["catalog_suffix"],
+            section_number=course["section_number"],
+        )
+    except Course.DoesNotExist:
+        try:
+            course_obj = Course.objects.create(
+                dept=course["teaching_area"],
+                catalog_number=course["catalog_number"] + course["catalog_suffix"],
+                section_number=course["section_number"],
+            )
+        except Exception as e:
+            log_error(
+                "An error occurred while creating a new course",
+                "core/logic.py check_or_create_course",
+                {"course_info": course},
+                e,
+            )
+            return None
+    return course_obj
+
+
+def create_user_course_association(user, course, yearterm):
+    # check if association already exists
+    associations = list(
+        UserCourse.objects.filter(
+            user_id=user.id, course_id=course.id, yearterm=yearterm
+        )
+    )
+    if associations:
+        return
+
+    try:
+        UserCourse.objects.create(
+            user_id=user.id, course_id=course.id, yearterm=yearterm
+        )
+    except Exception as e:
+        log_error(
+            "An error occurred while associating a user with a course",
+            "core/logic.py create_user_course_association",
+            {"user": user, "course": course, "yearterm": yearterm},
+            e,
+        )
+        return False
+    return True
```

Comment:
Looks like this can return None, True, or False. However, none of those values are ever used when you call this function.

Reply:
If it should return a boolean, and that will be used, then I would change the bare return to `return True`.